### ISSUE TITLE
wrong regex in CN matching

### ISSUE THREAD

**Issue by philfry** [matteocorti/check_ssl_cert#21]
Hi,

the CN matching (`-N`) fails when having additional informations, like an email address, in the subject.
```
$ openssl x509 -in mycert.pem -subject -noout -nameopt utf8,oneline,-esc_msb
subject= C = DE, ST = Foo, L = Bar, O = Baz, OU = yatta zonk, CN = my.host.name, emailAddress = me@address.invalid
$ openssl x509 -in mycert.pem -subject -noout -nameopt utf8,oneline,-esc_msb | sed -e "s/^.*[[:space:]]CN[[:space:]]=[[:space:]]//" -e "s/\/[[:alpha:]][[:alpha:]]*=.*\$//"
my.host.name, emailAddress = me@address.invalid
```
The slash is not a field delimiter, so the second sed instruction doesn't do anything. Better replace it with a simple `-e 's/,.*//'`:

```diff
--- nagios-plugins-ssl_cert-1.30.0/check_ssl_cert.cnmatch
+++ nagios-plugins-ssl_cert-1.30.0/check_ssl_cert
@@ -965,7 +965,7 @@ main() {
     DATE="$($OPENSSL x509 -in "${CERT}" -enddate -noout | sed -e "s/^notAfter=//")"
 
     CN="$($OPENSSL x509 -in "${CERT}" -subject -noout -nameopt utf8,oneline,-esc_msb |
-        sed -e "s/^.*[[:space:]]CN[[:space:]]=[[:space:]]//" -e "s/\/[[:alpha:]][[:alpha:]]*=.*\$//")"
+        sed -e "s/^.*[[:space:]]CN[[:space:]]=[[:space:]]//" -e "s/.*\$//")"
 
     CA_O="$($OPENSSL x509 -in "${CERT}" -issuer -noout | sed -e "s/^.*\/O=//" -e "s/\/[A-Z][A-Z]*=.*\$//")"
     CA_CN="$($OPENSSL x509 -in "${CERT}" -issuer -noout  | sed -e "s/^.*\/CN=//" -e "s/\/[A-Za-z][A-Za-z]*=.*\$//")"
```
kind regards

Philippe

**Comment by matteocorti** [comment]
The new regex deletes everything if the CN does not have additional information. For example

```
$ openssl x509 -subject -noout -nameopt utf8,oneline,-esc_msb -in ./www.ethz.ch.crt | sed -e "s/^.*[[:space:]]CN[[:space:]]=[[:space:]]//"
www.ethz.ch
$ openssl x509 -subject -noout -nameopt utf8,oneline,-esc_msb -in ./www.ethz.ch.crt | sed -e "s/^.*[[:space:]]CN[[:space:]]=[[:space:]]//" -e "s/.*\$//"

```

**Comment by philfry** [comment]
sorry, got a typo and missed the colon.
```diff
-        sed -e "s/^.*[[:space:]]CN[[:space:]]=[[:space:]]//" -e "s/\/[[:alpha:]][[:alpha:]]*=.*\$//")"
+        sed -e "s/^.*[[:space:]]CN[[:space:]]=[[:space:]]//" -e "s/,.*\$//")"
```

**Comment by matteocorti** [comment]
We need both

 * one regex to remove an additional email as in `CN = www.example.org/emailAddres=a@example.org`
 * and yours to remove additional elements (after a comma)

I will commit a fix and release a new version.

Do you have a public example I could include in the tests?

**Comment by philfry** [comment]
sure. With the patch from https://github.com/matteocorti/check_ssl_cert/issues/21#issuecomment-231967060 I get:
```
./check_ssl_cert -H mx.unixadm.org -p 25 -P smtp -N -d
[DBG] ROOT_CA = 
expect available (/usr/bin/expect)
timeout available (/usr/bin/timeout)
Perl module Date::Parse installed: enabling date computations
[DBG] check_ssl_version: 1.30.0
[DBG] OpenSSL binary: /usr/bin/openssl
[DBG] OpenSSL version: OpenSSL 1.0.1e-fips 11 Feb 2013
[DBG] System info: Linux monitor.unixadm.org 2.6.32-042stab108.8 #1 SMP Wed Jul 22 17:23:23 MSK 2015 x86_64 x86_64 x86_64 GNU/Linux
downloading certificate to /tmp
[DBG] executing with timeout (15s): echo -e 'QUIT\r' | /usr/bin/openssl s_client   -starttls smtp -connect mx.unixadm.org:25 -servername mx.unixadm.org -verify 6    2> /tmp/check_ssl_certKqa9l9 1> /tmp/check_ssl_certlteKMH
[DBG]   /usr/bin/timeout 15 /bin/sh -c "echo -e 'QUIT\r' | /usr/bin/openssl s_client   -starttls smtp -connect mx.unixadm.org:25 -servername mx.unixadm.org -verify 6    2> /tmp/check_ssl_certKqa9l9 1> /tmp/check_ssl_certlteKMH"
[DBG] storing a copy of the retrieved certificate in mx.unixadm.org.crt
[DBG] storing a copy of the OpenSSL errors in mx.unixadm.org.error
Parsing the certificate file
The certificate will expire in 183 day(s)
[DBG] check CN: mx.unixadm.org
[DBG] CN check finished
[DBG] OCSP: fetching issuer certificate  to /tmp/check_ssl_certvVJb80
[DBG] OCSP: storing a copy of the retrieved issuer certificate to 
cp: cannot create regular file ‘’: No such file or directory
[DBG] OCSP: host = 
openssl ocsp does not support the -header option: disabling OCSP checks
SSL_CERT OK - X.509 certificate for 'mx.unixadm.org' from 'Unixadm CA' valid until Jan 11 14:20:29 2017 GMT (expires in 183 days)|days=183;;;;
```
You'll probably need the `-s` switch as `mx.unixadm.org` is signed by a private CA.

**Comment by matteocorti** [comment]
Fixed in the release 1.31.0
